cucumber/kubernetes/features/step_definitions/authenticate_steps.rb

```diff
@@ -58,14 +50,15 @@ def authenticate_k8s(host, cert, key, conjur_id)
   end
 end
 
-Then(/^I( can)? authenticate pod matching "([^"]*)" with authn-k8s as "([^"]*)"( with prefix "([^"]*)")?( without cert and key)?$/) do |success, objectid, hostid_suffix, has_custom_prefix, hostid_prefix, nocertkey|
+Then(/^I( can)? authenticate pod matching "([^"]*)" with authn-k8s as "([^"]*)"(?: with prefix "([^"]*)")?( without cert and key)?$/) do |success, objectid, hostid_suffix, hostid_prefix, nocertkey|
   @request_ip ||= detect_request_ip(objectid)
 
-  if has_custom_prefix
-    conjur_id = "#{hostid_prefix}/#{hostid_suffix}"
-  else
-    conjur_id = conjur_resource_id(namespace, hostid_suffix)
-  end
+  conjur_id = 
+    if hostid_prefix
+      "#{hostid_prefix}/#{hostid_suffix}"
+    else
+      conjur_resource_id(namespace, hostid_suffix)
+    end
```

Comment:
This seems overly complicated. What about:
```ruby
conjur_id = "#{hostid_prefix}/#{hostid_suffix}"
if not hostid_prefix
  conjur_id = conjur_resource_id(namespace, hostid_suffix)
end
```

Reply:
Ha, I considered this.  This is Go style.  It's prettier, no doubt.  Both versions are clear.

But let's do some serious analysis.

I'll give you my reasons for choosing this and then let you do decide, since I was on the fence anyway. 

While "verbose" would be a fair accusation, I don't believe "overly complicated" is.  In fact, this is less complicated.  My version is a single _expression_ (if ... else can be expressions in ruby, as well as statements... ie, they can be just a long form of a ternary, which is how I would have written this if it fit on a line).  

Your rewrite, otoh, is a statement followed by a second conditional statement.  This highlights another difference: yours mutates a variable, the original is pure.  Idiomatic ruby tends to lean toward a functional style, whereas Go embraces procedural mutation.  I embrace the Go style when writing in Go, but prefer the ruby aesthetic overall.  

Finally, your version does "unnecessary work."  Ofc, practically speaking, this makes absolutely no difference.  But it bugged me on principle since with an expression you  can avoid it.  One might even argue it could be misleading.  I think it's a thin argument, but in the ruby world some developers would, indeed, ask: Why didn't you use an expression?

TLDR: We both react against it cause it's too long for what it is.  But it is the right thing -- it just happens that ruby style forces you to break up long ternaries like this:

https://rubystyle.guide/#no-multiline-ternary

Reply:
I think my issue with it is less about length (though it does play) but with the multiline ternary. Ternaries can in many cases complicate logic as-is. When you add that the result of the if/then block then rolls back _up_ into the assignment (that has the varname on 1 level higher line on top of that) it makes it pretty difficult to read and comprehend to me. I won't consider this a blocker but it requires a very Ruby-specific comprehension to read it fast so to me it makes it less maintainable.

Reply:
I very mildly disagree for reasons stated above but was happy to change given your feeling. 

Done!